test/test_linalg.py

```diff
@@ -2817,6 +2817,27 @@ def test_svd(self, device, dtype):
                 S_s = torch.svd(A, compute_uv=False).S
                 self.assertEqual(S_s, S)
 
+    @skipCUDAIfNoMagmaAndNoCusolver
+    @skipCPUIfNoLapack
+    @dtypes(torch.complex128)
+    def test_invariance_error_spectral_decompositions(self, device, dtype):
+        make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=True)
+        A = make_arg((3, 3))
+        with self.assertRaisesRegex(RuntimeError, "ill-defined"):
+            U, _, Vh = torch.linalg.svd(A, full_matrices=False)
+            (U + Vh).sum().backward()
```

Comment:
That reminds me of the condition check. Shall we have a float/double-dependent atol/rtol?

Reply:
I think it's good as it is. I wanted to put something that won't fire false positives, (hence the very lax condition in the `all_close`). I think the tolerances I put should be good. We just want to catch any violation of the invariance, which should give a tensor that's manifestly non-zero when the condition's tested.

Reply:
I see these thresholds are used for all the spectral decompositions. Seems quite arbitrary but all right...